inception-external-search-core/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/ExternalSearchHighlight.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019
+ * Ubiquitous Knowledge Processing (UKP) Lab
+ * Technische Universität Darmstadt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.inception.externalsearch;
+
+import java.io.IOException;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.common.base.Optional;
+import de.tudarmstadt.ukp.inception.support.annotation.OffsetSpan;
+
+/**
+ * This class represents a highlight from external search result.
+ * A highlight is a section of text containing occurrence(s) of keywords.
+ * The offsets tells where those keywords are within the original text document.
+ */
+public class ExternalSearchHighlight implements Serializable
+{
+
+    private static final long serialVersionUID = 4044755133816131842L;
+
+    private Optional<String> highlight;
+
+    private List<OffsetSpan> offsets;
+
+    public ExternalSearchHighlight(List<OffsetSpan> aOffsets) throws IOException
+    {
+        if (aOffsets.size() < 0) {
+            throw new IOException("Offsets must not be empty");
+        }
+        offsets = new ArrayList<>(aOffsets);
+
+    }
+
+    public ExternalSearchHighlight(String aHighlight, List<OffsetSpan> aOffsets) throws IOException
+    {
+        this(aOffsets);
+        highlight = Optional.fromNullable(aHighlight);
+    }
+
+    public String getHighlight()
```

Comment:
If you are going to use `Optional` at all, you want to return it, or you lose all the benefit of the type system forcing the users to account for the `null` case.

Reply:
Returning optional in such a case would be ok though.

Reply:
Oh, right, we can get `Optional` without Guava now.  (I spent years in a place stuck on Java 7, so I always forget this...). @shinyichen : Use `java.util.Optional`.

Reply:
Removed `Optional`

Reply:
You can (and should) still use `java.util.Optional` here for the return type only (not for field storage).